Subject:  Stop message spam from .cfi files
Patch Author:  Orson Teodoro
Date:  Wed Oct 13 11:45:22 PM PDT 2021 (Unix time:  1634193922)

License header notice for elf/readlib.c:

/* Copyright (C) 1999-2021 Free Software Foundation, Inc.
   This file is part of the GNU C Library.
   Contributed by Andreas Jaeger <aj@suse.de>, 1999 and
                  Jakub Jelinek <jakub@redhat.com>, 1999.

   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published
   by the Free Software Foundation; version 2 of the License, or
   (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License
   along with this program; if not, see <https://www.gnu.org/licenses/>.  */

The .cfi files indicate that the corresponding static or shared library has
been compiled with Clang CFI (Control Flow Integrity).

It silences these kinds of messages:
/sbin/ldconfig: File /usr/lib/libfoo.so.cfi is empty, not checked.

--- a/elf/readlib.c.orig	2021-08-01 18:33:43.000000000 -0700
+++ b/elf/readlib.c	2021-10-13 23:30:03.815533257 -0700
@@ -112,7 +112,10 @@ process_file (const char *real_file_name
       || (size_t) statbuf.st_size < sizeof (ElfW(Ehdr)))
     {
       if (statbuf.st_size == 0)
-	error (0, 0, _("File %s is empty, not checked."), file_name);
+	{
+	  if (strstr (file_name, ".cfi") == NULL)
+	    error (0, 0, _("File %s is empty, not checked."), file_name);
+	}
       else
 	{
 	  char buf[SELFMAG];
